### PR TITLE
manager: add support for token update events

### PR DESCRIPTION
**Pull request by tarcisiozf**
### Description:
By adding this support, we can allow an external plugin to renew tokens automatically

### Breaking changes:
- [ ]

### New features:
- [ ]

### Fixes:
- [ ]

### PR status:
- [ ] Version bumped
- [ ] Change-log updated
- [ ] Tests added or updated
- [ ] Documentation updated

### CODE REVIEW

File: lib/manager.js [hunk 1]
```diff
@@ -86,16 +86,40 @@ class Manager extends EventEmitter {
     this.wsPool = []
     this.authArgs = { dms, calc }
 
+    this._createRestClient()
+  }
+
+  _createRestClient () {
     this.rest = new RESTv2({
```
Comment (robertkowalski) [inline lib/manager.js]: hm, is the rest client used at all in `bfx-api-node-core`? else we might even be able to remove it...

could be also a separate PR in case we can remove it...

Comment (tarcisiozf) [reply lib/manager.js]: I'll take a look

Comment (tarcisiozf) [reply lib/manager.js]: not used by `api-node-core` nor `hf-algo`, I've removed it

File: package.json [hunk 2]
```diff
@@ -39,13 +39,18 @@
...
   },
   "dependencies": {
     "bfx-api-node-models": "^1.0.12",
     "bfx-api-node-rest": "^1.1.3",
     "bfx-api-node-util": "^1.0.2",
     "bfx-api-node-ws1": "^1.0.0",
-    "utf-8-validate": "^5.0.2"
+    "debug": "^4.3.1",
+    "lodash": "^4.17.21",
+    "utf-8-validate": "^5.0.2",
```
Comment (robertkowalski) [inline package.json]: Unused dependencies
* utf-8-validate

File: lib/manager.js [hunk 3]
```diff
@@ -309,6 +315,9 @@ class Manager extends EventEmitter {
     ev.on('exec:flags:set', onSetFlags.bind(this, this, { id }))
     ev.on('exec:ping', onPingEvent.bind(this, this, id))
 
+    // auth
+    ev.on('auth:token:updated', this._onAuthTokenUpdated.bind(this))
```
Comment (ghost) [inline lib/manager.js]: `auth:token:updated` is an internal Bitfinex event? Where is this documented?

Comment (tarcisiozf) [reply lib/manager.js]: it's a plugin-generated event, check the other MR. I'm also accepting suggestions of better ways to publish/consume it.

Comment (tarcisiozf) [reply lib/manager.js]: Maybe changing the `notifyPlugin` function and pass down the ref for the manager instance, so I can publish it directly. WDYT?

Comment (ghost) [reply lib/manager.js]: no, I think an event is better for low coupling, but need to be well documented

Comment (ghost) [reply lib/manager.js]: but.. if `auth:token:updated` is a plugin specific event, then the core depends on its plugin. @robertkowalski what do you think?

Can the plugin directly call the commands from `_onAuthTokenUpdated` (looks like only one `auth` method)?  This way we could get rid of `_onAuthTokenUpdated` method and event in the core

Comment (tarcisiozf) [reply lib/manager.js]: done, I've moved the call to the plugin

**Review by ghost — APPROVED** [review]
lgtm

File: test/lib/manager.js [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+          pluginThatMatches,
+          pluginDoesNoMatch
+        ]
+      })
+      instance.notifyPlugin = notifyPluginStub
+      instance.notifyPlugins(type, section, name, args)
+
+      assert.calledOnce(notifyPluginStub)
+      assert.calledWithExactly(notifyPluginStub, pluginThatMatches, section, name, args)
+      expect(notifyPluginStub.calledWith(pluginDoesNoMatch)).to.be.false
```
Comment (robertkowalski) [inline test/lib/manager.js:151]: great test coverage, thanks!

**Review by robertkowalski — APPROVED** [review]
lgtm
